os/kernel/task/task_prctl.c

```diff
@@ -347,6 +351,28 @@ int prctl(int option, ...)
 		va_end(ap);
 		return ret;
 	}
+#endif
+#ifdef CONFIG_SYSTEM_REBOOT_REASON
+	case PR_REBOOT_REASON_READ:
+	{
+		va_end(ap);
+		return reboot_reason_read();
+	}
+	case PR_REBOOT_REASON_WRITE:
+	{
+		reboot_reason_code_t reason = (reboot_reason_code_t)va_arg(ap, int);
+
+		reboot_reason_write(reason);
+
+		va_end(ap);
+		return OK;
+	}
+	case PR_REBOOT_REASON_CLEAR:
+	{
+		reboot_reason_clear();
```

Comment:
ditto, else make function return void.

Reply:
I modified the prototype as void return. Thanks.